server/events.go

```diff
@@ -600,28 +600,28 @@ func (s *Server) initEventTracking() {
 
 	monSrvc := map[string]msgHandler{
 		"VARZ": func(sub *subscription, _ *client, subject, reply string, msg []byte) {
-			optz := &VarzOptions{}
-			s.zReq(reply, msg, optz, func() (interface{}, error) { return s.Varz(optz) })
+			optz := &VarzEventOptions{}
+			s.zReq(reply, msg, &optz.EventFilterOptions, optz, func() (interface{}, error) { return s.Varz(&optz.VarzOptions) })
```

Comment:
Do you foresee even filter options to be different based on the event options? If not, then I would not pass them here but on in zReq() invoke `filterRequest(*optz.EventFilterOptions)`

Reply:
They are passed here so inside zReq I have the options after they are parsed. 
This allows me to avoid the dedicated parsing step in filterRequest.

Reply:
Right, because you pass `optz` as `interface{}`.